### PR TITLE
*: fix load statement can not handle the upper case user variable

### LINKED ISSUE #34420: load statement with user variable cannot handle the upper case.
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```
LOAD DATA INFILE 'file.txt'
  INTO TABLE t1
  (column1, @VAR1)
  SET column2 = @VAR1/100;
```

The problem is that:
When setting the to variable into a map, TiDB uses the original text.
But when getting the variable from the map, TiDB uses the lower case text.

### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by wjhuang2016**
Signed-off-by: wjhuang2016 <huangwenjun1997@gmail.com>

### What problem does this PR solve?

Issue Number: close #34420

Problem Summary:
When setting the to variable into a map, TiDB uses the original text.
But when getting the variable from the map, TiDB uses the lower case text.

### What is changed and how it works?
Store the lower case variable name instead of the original name.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
fix load statement can not handle the upper case user variable
```

### CODE REVIEW

File: expression/builtin_other.go [hunk 1]
```diff
@@ -776,7 +776,6 @@ func (b *builtinSetStringVarSig) evalString(row chunk.Row) (res string, isNull b
 	if err != nil {
 		return "", isNull, err
 	}
-	varName = strings.ToLower(varName)
```
Comment (cfzjywxk) [inline expression/builtin_other.go:779]: Will this break any implicit dependencies and introduce regression?

Comment (wjhuang2016) [reply expression/builtin_other.go:779]: We assume that the lower case is used, that's what we have done when getting the variables.
So wrapping strings.ToLower(varName) into SetUserVar would avoiding the missing converstion.

Comment (wjhuang2016) [reply expression/builtin_other.go:779]: The reason of introducing this bug is that:
1. first, convert to the lower-case name, then call SetUserVar.
2. when supporting user variable for load statement, call SetUserVar directly.

So I move the `convert to the lower-case name` into SetUserVar.

**Review by tiancaiamao — APPROVED** [review]

**Pull request merged**